File: clojure/word-count/src/word_count.clj
(ns word-count
  (:require [clojure.string :as str]))

(defn reducer [acc word]
  (update acc word #(if (nil? %1) 1 (inc %1))))

(defn word-count [s]
  (let [words (str/split (str/lower-case s) #"[^A-Za-z0-9]+")]
  (reduce reducer {} words)))

File: python/word-count/word_count.py
import re
from collections import Counter

IGNORED = "[^a-z0-9' ]"
SEPARATOR = " "

def gen(words):
   for word in words:
        if not word:
            continue
        if word[0] == "'":
            word = word[1:-1]
        yield word

def count_words(sentence):
    lower_sentence = sentence.lower()
    words = re.sub(IGNORED, SEPARATOR, lower_sentence).split(SEPARATOR)

    result = Counter(gen(words))

    return result
